File: Python/Programmers/table_hash_function.py
def solution(data, col, row_begin, row_end):
    sorted_data = sorted(data, key=lambda x : (x[col-1], -x[0]))
    sums = []
    for i in range(row_begin, row_end+1):
        sum = 0
        for d in sorted_data[i-1]:
            sum += d % i
        sums.append(sum)
        
    answer = sums[0]
    for i in sums[1:]:
        answer = answer ^ i
        
    return answer

File: Java/Programmers/table_hash_function.java
package Programmers;

import java.util.Arrays;

public class table_hash_function {
    public int solution(int[][] data, int col, int row_begin, int row_end) {
        int answer = 0;
        Arrays.sort(data,(o1, o2) -> o1[col-1] != o2[col-1] ? o1[col-1]-o2[col-1] : o2[0]-o1[0]);
        for(int i = row_begin; i < row_end+1; i++){
            int sum = 0;
            for(int d : data[i-1]){
                sum += d % i;
            }
            answer = answer^sum;

        }
        return answer;
    }
}
